app/uk/gov/hmrc/helptosave/models/EligibilityCheckResult.scala

```diff
@@ -16,14 +16,33 @@
 
 package uk.gov.hmrc.helptosave.models
 
-import play.api.libs.json.{Format, Json}
+import play.api.libs.json.Json._
+import play.api.libs.json._
 
-/**
-  * The result of performing an eligibility check. If the user is eligible
-  * return their details, otherwise return [[None]]
-  */
-case class EligibilityCheckResult(result: Option[UserInfo])
+case class EligibilityCheckResult(result: Either[HtsError, UserInfo])
 
 object EligibilityCheckResult {
-  implicit val eligibilityResultFormat: Format[EligibilityCheckResult] = Json.format[EligibilityCheckResult]
+
+  val reads: Reads[EligibilityCheckResult] = new Reads[EligibilityCheckResult] {
+    override def reads(json: JsValue): JsResult[EligibilityCheckResult] = {
+
+      val result = (json \ "result").get
+
+      val reads = result.asOpt[HtsError].fold(
+        EligibilityCheckResult(Right(result.as[UserInfo]))
+      )(htsError ⇒ EligibilityCheckResult(Left(htsError)))
+
+      JsSuccess(reads)
+    }
+  }
+
+  val writes: Writes[EligibilityCheckResult] = new Writes[EligibilityCheckResult] {
+    override def writes(o: EligibilityCheckResult): JsValue = Json.obj(
+      o.result.fold(
+        errors ⇒ "result" -> Json.toJson(errors),
+        maybeUserInfo ⇒ "result" -> Json.toJson(maybeUserInfo)
+      ))
+  }
+
+  implicit val eligibilityResultFormat: Format[EligibilityCheckResult] = Format(reads, writes)
```

Comment:
You could have written
```scala
val format: Format[EligibilityCheckResult] = new Format {
 ...
}
```
instead of writing the reads and writes separately and combining them again